storage/sql/config_test.go

```diff
@@ -77,6 +77,55 @@ func getenv(key, defaultVal string) string {
 
 const testPostgresEnv = "DEX_POSTGRES_HOST"
 
+func TestCreateDataSourceName(t *testing.T) {
+	var testCases = []struct {
+		description string
+		input       *Postgres
+		expected    string
+	}{
+		{
+			description: "with no configuration, connect_timeout is specified with default value",
+			input: &Postgres{
+				SSL: PostgresSSL{
+					Mode: "disable",
+				},
+			},
+			expected: "connect_timeout=0 sslmode='disable'",
+		},
+		{
+			description: "with unix socket host",
+			input: &Postgres{
+				Host: "/var/run/postgres",
+				SSL: PostgresSSL{
+					Mode: "disable",
+				},
+			},
+			expected: "connect_timeout=0 host='/var/run/postgres' sslmode='disable'",
+		},
+		{
+			description: "with tcp host",
+			input: &Postgres{
+				Host: "coreos.com",
+				SSL: PostgresSSL{
+					Mode: "disable",
+				},
+			},
+			expected: "connect_timeout=0 host='coreos.com' sslmode='disable'",
+		},
```

Comment:
👏 nice set of tests. Could you add one that requires the escaping, please? Could be a password. 😉 Would be best if it included both a `'` and some `\` 😄 

Also, and I hope I'm not asking for too much, including a test case with `sslmode != "disable"` would seem worthwhile, too, wouldn't it?

Reply:
Ah good point. I should have caught that! Adding more tests now.

I may be rushing these PRs a bit. Most of these PRs are for code written many moons ago by other folks, so I'll start reviewing the changes more thoroughly before submitting.